app/renderer/components/Inspector/HighlighterRects.js

```diff
@@ -1,20 +1,165 @@
 import React, { Component } from 'react';
 import HighlighterRect from './HighlighterRect';
+import HighlighterCentroid from './HighlighterCentroid';
 import B from 'bluebird';
-import { SCREENSHOT_INTERACTION_MODE } from './shared';
+import { SCREENSHOT_INTERACTION_MODE, parseCoordinates, RENDER_CENTROID_AS } from './shared';
 
 const {TAP, SWIPE, SELECT} = SCREENSHOT_INTERACTION_MODE;
+const {CENTROID, OVERLAP, EXPAND} = RENDER_CENTROID_AS;
+
+// Centroid positions are shifted based off screenshot width % for perfect alignment
+const POS_OFFSET = 0.03;
+// Distance in screenshot-width % for overlapping centroids from +/- centroids
+const DIST_OFFSET = 0.06;
 
 /**
  * Shows screenshot of running application and divs that highlight the elements' bounding boxes
  */
 export default class HighlighterRects extends Component {
 
+  getElements (source) {
+    const elementsByOverlap = this.buildElementsWithProps(source, null, [], {});
+    let elements = [];
+
+    for (const key of Object.keys(elementsByOverlap)) {
+      if (elementsByOverlap[key].length > 1) {
+        const {centerX, centerY} = elementsByOverlap[key][0].properties;
+
+        // Create new element obj which will be a +/- centroid
+
+        const element = {
+          type: EXPAND,
+          element: null,
+          parent: null,
+          properties: {
+            left: null,
+            top: null,
+            width: null,
+            height: null,
+            centerX,
+            centerY,
+            path: key,
+            keyCode: key,
+            container: null,
+            accessible: null
+          }
+        };
+        elements = elements.concat(element,
+          this.updateElementsCoords(elementsByOverlap[key], key));
+      } else {
+        elements.push(elementsByOverlap[key][0]);
```

Comment:
is the first element guaranteed to exist?

Reply:
Yup its guaranteed to exist because all the elements in the xml file are sorted based off of their positions, and so any element that has a unique position is added to an array and assigned a new object so an object cannot have an empty array.